jpos/src/main/java/org/jpos/iso/ISOServer.java

```diff
@@ -654,31 +654,23 @@ public ISOChannel getLastConnectedISOChannel () {
      * @return ISOChannel under the given name
      */
     public ISOChannel getISOChannel (String name) {
-        WeakReference ref = (WeakReference) channels.get (name);
-        if (ref != null) {
-            return (ISOChannel) ref.get ();
-        }
-        return null;
-    }
+        WeakReference<ISOChannel> ref = channels.get(name);
+        if (ref == null)
+            return null;
 
+        return ref.get();
+    }
 
     @Override
     public String getISOChannelNames () {
-        StringBuilder sb = new StringBuilder ();
-        Iterator iter = channels.entrySet().iterator();
-        for (int i=0; iter.hasNext(); i++) {
-            Map.Entry entry = (Map.Entry) iter.next();
-            WeakReference ref = (WeakReference) entry.getValue();
-            ISOChannel c = (ISOChannel) ref.get ();
-            if (c != null && !LAST.equals (entry.getKey()) && c.isConnected()) {
-                if (i > 0) {
-                    sb.append (' ');
-                }
-                sb.append (entry.getKey());
-            }
-        }
-        return sb.toString();
+        return channels.entrySet().stream()
+                .filter(ent -> !LAST.equals(ent.getKey()))
+                .filter(ent -> ent.getValue().get() != null)
+                .filter(ent -> ent.getValue().get().isConnected())
```

Comment:
Very rare to occur, but in theory, I think we can have a race condition between line 668 and 669, because value is a WeakReference.